python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -492,12 +495,29 @@ def external_asset_graph_from_defs(
                         )
                         all_upstream_asset_keys.add(upstream_asset_key)
 
-    source_asset_keys = all_upstream_asset_keys.difference(node_defs_by_asset_key.keys())
+    asset_keys_without_definitions = all_upstream_asset_keys.difference(
+        node_defs_by_asset_key.keys()
+    ).difference(foreign_assets_by_key.keys())
```

Comment:
nit: Maybe we can throw an error when a repository has a collision between asset nodes and foreign asset keys?

Maybe we should also consider having some sort of hash so that we can error even when we have different asset nodes across different jobs with the same asset keys.

Reply:
Good point. I updated to throw an error when a repository has a collision between asset nodes and foreign asset keys.